src/engine/objects/SimPyramid.ts

```diff
@@ -27,25 +27,38 @@ export class SimPyramid extends SimObject {
     const color = spec.baseColor ? spec.baseColor : DEFAULT_PYRAMID_COLOR;
     const initialPosition: Vector2d = { x: 0, y: 0 };
 
-    const pyramidGeom = new THREE.Geometry();
     const halfX = spec.baseDimensions.x / 2;
     const halfZ = spec.baseDimensions.y / 2;
-    pyramidGeom.vertices = [
+
+    const pyramidGeom = new THREE.BufferGeometry();
+    pyramidGeom.setFromPoints([
       new THREE.Vector3(-halfX, 0, -halfZ),
       new THREE.Vector3(-halfX, 0, halfZ),
       new THREE.Vector3(halfX, 0, halfZ),
       new THREE.Vector3(halfX, 0, -halfZ),
       new THREE.Vector3(0, spec.height, 0),
-    ];
-    pyramidGeom.faces = [
-      new THREE.Face3(0, 2, 1),
-      new THREE.Face3(0, 3, 2),
-      new THREE.Face3(1, 4, 0),
-      new THREE.Face3(2, 4, 1),
-      new THREE.Face3(3, 4, 2),
-      new THREE.Face3(0, 4, 3),
-    ];
-    pyramidGeom.computeFaceNormals();
+    ]);
+    pyramidGeom.setIndex([
```

Comment:
Could https://threejs.org/docs/?q=Face#examples/en/math/convexhull/Face be used as a replacement for `Face3`? I am wondering if that would be slightly more readable. I checked the release notes where they announced this breaking change, but didn't find any mention of a replacement type that should be used.

Reply:
Hmm, that looks to be part of the Convex Hull maths stuff rather than for rendering stuff. (at least I gather that from the location). The `BufferGeometry` only has this `setIndex` function for setting the face indexes.

Reply:
Understood!